### ISSUE TITLE
Crash while creating aws_cloudfront_distribution

### ISSUE THREAD

**Issue by schammah** [terraform-providers/terraform-provider-aws#12098]
### Terraform Version

```terraform v0.12.20
+ provider.aws v2.49.0```

### Affected Resource(s)

* aws_cloudfront_distribution

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
locals {
  dist_name = "${upper(var.name_prefix)} CDN"
}

data "aws_region" "current" {}

module "cloudfront_dist" {
  source  = "jmgreg31/cloudfront/aws"
  version = "4.2.0"
  // for local_dev source = "../../../terraform-aws-cloudfront"

  create_cf           = var.create_cf
  default_root_object = var.cf_default_root_object
  comment             = local.dist_name
  price               = var.cf_price_class

  dynamic_logging_config = {
    bucket = var.s3_accesslogs_bucket_dn
    prefix = "cf_access_logs/"
  }

  minimum_protocol_version = "TLSv1.2_2018"
  # alias = [var.aliases]
  #acm_certificate_arn = "take variable from data in fe_stack"

  dynamic_s3_origin_config = concat([{
    domain_name            = var.s3_deploy_bucket_dn
    origin_id              = var.s3_deploy_bucket_arn
    origin_path            = var.s3_origin_path == null ? null : "/${var.s3_origin_path}"
    origin_access_identity = var.s3_origin_access_identity
  }], var.cf_additional_s3_origin_config)

  dynamic_custom_origin_config = var.cf_custom_origin_config

  dynamic_default_cache_behavior = {
    allowed_methods             = ["GET", "HEAD"]
    cached_methods              = ["GET", "HEAD"]
    target_origin_id            = var.s3_deploy_bucket_arn
    query_string                = false
    cookies_forward             = "none"
    viewer_protocol_policy      = "redirect-to-https"
    min_ttl                     = 31536000
    default_ttl                 = 31536000
    max_ttl                     = 31536000
    lambda_function_association = var.cf_lambda_default_behavior
  }

  dynamic_ordered_cache_behavior = var.cf_ordered_cache_behavior

  tag_name        = local.dist_name
  additional_tags = var.adsk_tags
}
```
`variables.tf`
```
variable create_cf {
  type        = bool
  description = "Specify wether a Cloudfront Distribution should be created as well or not"
  default     = false
}

variable name_prefix {
  type        = string
  description = "Name Prefix to use in Cloudfront CDN Module"
}

variable cf_price_class {
  type        = string
  description = "The price class for this distribution. One of PriceClass_All, PriceClass_200, PriceClass_100"
  default     = "PriceClass_100"
}

variable cf_default_root_object {
  description = "The object that you want CloudFront to return (for example, index.html) when an end user requests the root URL"
  type        = string
  default     = null
}

variable s3_origin_path {
  type        = string
  description = "S3 Prefix/Folder Destination, if provided will be configured as the Origin Path on the Default S3 Origin"
  default     = null
}

variable s3_origin_access_identity {
  type        = string
  description = "S3 Origin Access ID configured to CF Default Origin in order to allow its access to the S3 Depliy Bucket"
}

variable s3_accesslogs_bucket_dn {
  type        = string
  description = "S3 AccessLogs Bucket Domain Name"
}

variable s3_deploy_bucket_dn {
  type        = string
  description = "S3 Deploy Bucket Domain Name"
}

variable s3_deploy_bucket_arn {
  type        = string
  description = "S3 Deploy Bucket ID"
}

variable cf_additional_s3_origin_config {
  type        = list(map(string))
  description = "Additional S3 to be configured as Origins in the Cloudfront Distribution"
  default     = []
}

variable cf_custom_origin_config {
  type        = any
  description = "Additional S3 to be configured as Origins in the Cloudfront Distribution"
  default     = {}
}

variable cf_lambda_default_behavior {
  type        = list(map(string))
  description = "Lambda@Edge Functions to associate to the Default Cache Behavior "
  default     = []
}

variable cf_ordered_cache_behavior {
  type        = any
  description = "Addional Cache Behaviors to configure for the Cloudfront Distribution"
  default     = []
}

variable adsk_tags {
  type        = map
  description = "Map of defined Autodesk Tags"
  default     = {}
}

```
[modules url](https://github.com/jmgreg31/terraform-aws-cloudfront)

### Debug Output

```
Error: Provider produced inconsistent final plan

When expanding the plan for
module.fe-stack.module.cf_distribution.module.cloudfront_dist.aws_cloudfront_distribution.cloudfront_distribution[0]
to include new values learned so far during apply, provider
"registry.terraform.io/-/aws" produced an invalid new value for .origin:
planned set element
cty.ObjectVal(map[string]cty.Value{"custom_header":cty.SetVal([]cty.Value{cty.ObjectVal(map[string]cty.Value{"name":cty.UnknownVal(cty.String),
"value":cty.UnknownVal(cty.String)})}),
"custom_origin_config":cty.ListValEmpty(cty.Object(map[string]cty.Type{"http_port":cty.Number,
"https_port":cty.Number, "origin_keepalive_timeout":cty.Number,
"origin_protocol_policy":cty.String, "origin_read_timeout":cty.Number,
"origin_ssl_protocols":cty.Set(cty.String)})),
"domain_name":cty.UnknownVal(cty.String),
"origin_id":cty.UnknownVal(cty.String),
"origin_path":cty.UnknownVal(cty.String),
"s3_origin_config":cty.ListVal([]cty.Value{cty.ObjectVal(map[string]cty.Value{"origin_access_identity":cty.UnknownVal(cty.String)})})})
does not correlate with any element in actual.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.


Error: Provider produced inconsistent final plan

When expanding the plan for
module.fe-stack.module.cf_distribution.module.cloudfront_dist.aws_cloudfront_distribution.cloudfront_distribution[0]
to include new values learned so far during apply, provider
"registry.terraform.io/-/aws" produced an invalid new value for
.default_cache_behavior[0].lambda_function_association: planned set element
cty.ObjectVal(map[string]cty.Value{"event_type":cty.UnknownVal(cty.String),
"include_body":cty.UnknownVal(cty.Bool),
"lambda_arn":cty.UnknownVal(cty.String)}) does not correlate with any element
in actual.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.
```
### Expected Behavior

Run and create the resource on the 1st iteration run

### Actual Behavior
Crash on the first run,
Runs fine on the second run only

### Steps to Reproduce

1. `terraform apply`

**Comment by schammah** [comment]
this is a duplicate of #10088 probably

**Comment by schammah** [comment]
closing as this is solved with 0.13.1 or 0.13.2 and further
